### PR TITLE
Copy Python tests for Modulemd.ModuleStream.tracker into C

### PR DESCRIPTION
**Pull request by srikavin**
This is related to #199. I also added Unicode tests.

### CODE REVIEW

File: modulemd/tests/test-modulemd-modulestream.c [hunk 1]
```diff
@@ -331,6 +332,136 @@ module_stream_v2_test_documentation (ModuleStreamFixture *fixture,
...
+
+  g_object_get (stream, MMD_TEST_TRACKER_PROP, &tracker_prop, NULL);
+  tracker = modulemd_module_stream_v1_get_tracker (stream);
+
+  g_assert_null (tracker);
+  g_assert_null (tracker_prop);
+
+  g_clear_pointer (&tracker_prop, g_free);
+
+  // Test Unicode values
```
Comment (OrionStar25) [inline modulemd/tests/test-modulemd-modulestream.c]: According to the docs, the `tracker` is:
`The website address of the upstream bug tracker for this module.`

So it doesn't really make sense to have Unicode characters in a website URL. Now we should either remove this test assuming no one is ever going to add a URL containing Unicode or handle this case in the main function by returning `NULL` whenever we encounter an invalid URL.

@sgallagher What do you suggest?

File: modulemd/tests/test-modulemd-modulestream.c [hunk 2]
```diff
@@ -331,6 +332,136 @@ module_stream_v2_test_documentation (ModuleStreamFixture *fixture,
...
+
+  g_object_get (stream, MMD_TEST_TRACKER_PROP, &tracker_prop, NULL);
+  tracker = modulemd_module_stream_v2_get_tracker (stream);
+
+  g_assert_null (tracker);
+  g_assert_null (tracker_prop);
+
+  g_clear_pointer (&tracker_prop, g_free);
+
+  // Test Unicode values
```
Comment (OrionStar25) [inline modulemd/tests/test-modulemd-modulestream.c]: Same as above.

File: modulemd/tests/test-modulemd-modulestream.c (hunk 1 above)
Comment (sgallagher) [reply modulemd/tests/test-modulemd-modulestream.c]: Leave it as-is for now. There's nothing *stopping* someone right now from putting unicode into the metadata here. We should probably look at using http-parser or llhttp to validate that we have validly-formatted URLs though. Mind opening a ticket to track that?

**Comment by srikavin** [comment]
@sgallagher I've rebased this branch onto master.

**Pull request merged**